lang/cs/Org.Apache.REEF.Client/YARN/RESTClient/YarnClientNoCancellationToken.cs

```diff
@@ -47,5 +47,11 @@ internal sealed partial class YarnClient
             await new RemoveSynchronizationContextAwaiter();
             return await CreateNewApplicationAsync(CancellationToken.None);
         }
+
+        public async Task<Application> SubmitApplicationAsync(SubmitApplication submitApplicationRequest)
+        {
+            await new RemoveSynchronizationContextAwaiter();
```

Comment:
It is a general purpose helper, that can be used even in tight loops doing async operation. We do not want to increased GC pressure there.
Further take a look at TaskAwaiter https://msdn.microsoft.com/en-us/library/hh138386(v=vs.110).aspx
It is the same thing that base class library does.